sentinel-2/evi/script.js

```diff
@@ -1,27 +1,42 @@
+//VERSION=3
 // Enhanced Vegetation Index  (abbrv. EVI)
 // General formula: 2.5 * (NIR - RED) / ((NIR + 6*RED - 7.5*BLUE) + 1)
 // URL https://www.indexdatabase.de/db/si-single.php?sensor_id=96&rsindex_id=16
+function setup() {
+   return {
+      input: ["B02", "B04", "B08", "dataMask"],
+      output: { bands: 4 }
+   };
+}
 
-let EVI = 2.5 * (B08 - B04) / ((B08 + 6.0 * B04 - 7.5 * B02) + 1.0);
+const ramp = [
+   [-0.5, 0x0c0c0c],
+   [-0.2, 0xbfbfbf],
+   [-0.1, 0xdbdbdb],
+   [0, 0xeaeaea],
+   [0.025, 0xfff9cc],
+   [0.05, 0xede8b5],
+   [0.075, 0xddd89b],
+   [0.1, 0xccc682],
+   [0.125, 0xbcb76b],
+   [0.15, 0xafc160],
+   [0.175, 0xa3cc59],
+   [0.2, 0x91bf51],
+   [0.25, 0x7fb247],
+   [0.3, 0x70a33f],
+   [0.35, 0x609635],
+   [0.4, 0x4f892d],
+   [0.45, 0x3f7c23],
+   [0.5, 0x306d1c],
+   [0.55, 0x216011],
+   [0.6, 0x0f540a],
+   [1, 0x004400],
+];
 
-if (EVI<-1.1) return [0,0,0];
```

Comment:
There was a discussion about using visualiser or conditional statement in the past with Monja and Anja and they decided to go with conditional statement as it was eaiser for users to understand and adapt (see [here](https://git.sinergise.com/sh-vas/sh-vas/-/issues/1276#note_536205)).

But I do think it's better to use a visualiser. Maybe we can raise this again in the eob channel or cdse channel?

Reply:
I think it's time to move on from those bulky if elseif construct. Just from a maintainability standpoint